walkthrough-2 (Portfolio)

dmosc opened the pull request.

dmosc committed:
walkthrough-1 (Initial setup) Set initial README.md \nChanged initial README.md content to personalize repo.

dmosc committed:
walkthrough-2 (Webapp) Add initial portfolio section structure.

+ Decided to rely on a Bootstrap CDN server to style the portfolio.

+ Required imports where added at the beginning of index.html.

+ Added as well the general structure of the portfolio consisting of a
3 column grid.

+ Created CSS container classes in style.css to structure and format
each section.

*** No breaking changes ***

dmosc committed:
walkthrough-2 (Elements insertion) Add JS functions to append HTML elements with portfolios' information dynamically.

+ Created hobbies.js to append all my hobbies into the hobbies section inside index.html. Hobbies are locally
stored as an array.

+ Created projects.js to append all my projects into the projects section inside index.html. Projects are locally
stored as an array holding objects.

+ Created social_networks.js to append all my links into the social_networks section inside index.html. The
social networks are stored as an object.

+ The index.js file contains as imports the rest of the js files. This way the other files execute when
index.js is imported. This file is imported as a JS module inside index.html to allow this format.

*** No breaking changes ***

dmosc committed:
walkthrough-w-2 (Static files) Add profile_picture static image.

+ Uploaded profile_picture.jpg image to load it inside the portfolio.

*** No breaking changes ***

anniefu commented on portfolio/src/main/webapp/static/constants/projects.js:
```diff
@@ -0,0 +1,33 @@
+        const project_to_append = document.createElement("li");
+
+        project_to_append.classList.add("list-group-item", "text-wrap");
+        project_to_append.style = "width: 100%;";
```
It'd probably be best to consolidate all your styling to your .css files. That way in the future if someone was trying to adjust something, it's more transparent where the styling is coming from.

I'd recommend adding a class/id to your project_to_append element that your CSS file can use.

dmosc replied:
That's a really good suggestion. That approach centralizes better the CSS styling.
I'll go ahead and resolve it in a new commit.

anniefu commented on portfolio/src/main/webapp/static/constants/social_networks.js:
```diff
@@ -0,0 +1,24 @@
+
+        const icon = document.createElement("i");
+        icon.classList.add('fab', `fa-${key}`);
+        icon.style = "width: 20px; height: 20px;";
```
Same recommendation as above, to put the styling in the CSS file.

anniefu reviewed (approved):
Hey Oscar, good start!

I left a suggestion that I think would be good to take action on, but aren't critical. I'm gonna give approval right now so that if you push a new commit to fix them, you can merge immediately after.

ssmall reviewed (approved):
Looks good! I second Annie's comments about styling.

dmosc committed:
walkthrough-2 (Styles) Extrapolate CSS styles inside static/constants.

+ Some HTML injected objects inside the JS functions where being manually
styled, instead of grouping styles into CSS classes. Created classes, and
instead of directly adding styling rules inside JS documents, the HTML
elements now get the class reference.

*** No breaking changes ***
